test/unit/sdam/server_description.test.ts

```diff
@@ -0,0 +1,156 @@
+        lhs: new ServerDescription('127.0.0.1:27017', undefined, { error: new TypeError('test') }),
+        rhs: new ServerDescription('127.0.0.1:27017', undefined, { error: undefined }),
+        equal: false
+      }
+    ];
+
+    for (const test of errorEqualityTests) {
+      it(test.description, function () {
+        expect(test.lhs.equals(test.rhs)).to.equal(test.equal);
+      });
+    }
+  });
+
+  it('should sensibly parse an ipv6 address', function () {
+    const description = new ServerDescription('[ABCD:f::abcd:abcd:abcd:abcd]:27017');
+    expect(description.host).to.equal('abcd:f::abcd:abcd:abcd:abcd');
+    expect(description.port).to.equal(27017);
+  });
+
+  describe('compareTopologyVersion()', () => {
+    const processIdZero = new ObjectId('00'.repeat(12));
+    const processIdOne = new ObjectId('00'.repeat(11) + '01');
+    type CompareTopologyVersionTest = {
+      title: string;
+      currentTv?: TopologyVersion;
+      newTv?: TopologyVersion;
+      out: 0 | -1 | 1;
+    };
+
+    const compareTopologyVersionEqualTests: CompareTopologyVersionTest[] = [
+      {
+        title: 'when process ids are equal and both counter values are Long.ZERO',
+        currentTv: { processId: processIdZero, counter: Long.ZERO },
+        newTv: { processId: processIdZero, counter: Long.ZERO },
+        out: 0
+      },
+      {
+        title: 'when process ids are equal and both counter values are non-zero numbers',
+        // @ts-expect-error: Testing that the function handles numbers
+        currentTv: { processId: processIdZero, counter: 2 },
+        // @ts-expect-error: Testing that the function handles numbers
+        newTv: { processId: processIdZero, counter: 2 },
+        out: 0
+      },
+      {
+        title: 'when process ids are equal and both counter values are zero numbers',
+        // @ts-expect-error: Testing that the function handles numbers
+        currentTv: { processId: processIdZero, counter: 0 },
+        // @ts-expect-error: Testing that the function handles numbers
+        newTv: { processId: processIdZero, counter: 0 },
+        out: 0
+      },
+      {
+        title:
+          'when process ids are equal and counter values are equal but current has a different type',
+        currentTv: { processId: processIdZero, counter: Long.fromNumber(2) },
+        // @ts-expect-error: Testing that the function handles numbers
+        newTv: { processId: processIdZero, counter: 2 },
+        out: 0
+      },
+      {
+        title:
+          'when process ids are equal and counter values are equal but new has a different type',
+        // @ts-expect-error: Testing that the function handles numbers
+        currentTv: { processId: processIdZero, counter: 2 },
+        newTv: { processId: processIdZero, counter: Long.fromNumber(2) },
+        out: 0
+      }
+    ];
+    const compareTopologyVersionLessThanTests: CompareTopologyVersionTest[] = [
+      {
+        title: 'when both versions are nullish',
```

Comment:
would it be worth having two more tests that have different kinds of nullish for completeness? (null with undefined, and undefined with null)

Reply:
Sure!